uwblueprint/supportive-housing
# Verification Page

## Notion task link


[Login/Signup Verification](https://www.notion.so/uwblueprintexecs/Login-Signup-Verification-43263134e4e14481a0de27c0c011a780)



## Implementation description
* Updated `AuthenticatedUser` type to include if the user is verified in `AuthTypes.tsx`
* Updated login and register in `auth_routes.py`
* Created `is_authorized_by_token` in `auth_service.py` to check if a user is verified using the access token
* Created `isVerified` in `AuthAPIClient.ts` for the verification page button to update the AuthenticatedUser context if the user becomes verified



## Steps to test
1. Access the database using `docker exec -it SHOW-database /bin/bash -c "psql -U postgres -d show_db"`
2. Invite a new account using `insert into users (first_name, last_name, role, user_status, email) values ('your first name', 'your last name', 'Admin', 'Invited', 'your email');`
3. Check if after the Signup page the user is taken to the verification page 
4. Check that pressing the button on the Verification page prompts the user with an error toast message
5. Verify account from email
6. Check that pressing the button on the Verification page takes the user to the homepage
7. Logout from the account
8. Sign in to account from Login page and check that the user is taken directly to the homepage 


## What should reviewers focus on?
* Should there be a button for the user to logout from the Verification page to prevent getting stuck if the verification email expires? 

## Screenshot
![image](https://github.com/uwblueprint/supportive-housing/assets/43042601/1e057fd3-1029-4d11-b389-33626a8bcd1a)

## Checklist
